src/MoveGenerator.java

```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class MoveGenerator {
+
+    private List<MoveInterface> possibleMoves;
+    private Board board;
+    private List<Square> currentPlayersOccupiedSquares;
+
+    public MoveGenerator(Board board, PieceColor color) {
+        this.possibleMoves = new ArrayList<MoveInterface>();
+        this.board = board;
+        currentPlayersOccupiedSquares = new ArrayList<Square>();
+        determineThisPlayersOccupiedSquares(color);
+    }
+
+    public List<MoveInterface> getPossibleMoves() {
+        determineNonJumpMoves();
+        determineJumpMoves();
+        return possibleMoves;
+    }
+
+    private void determineThisPlayersOccupiedSquares(PieceColor color) {
+        for (Square square : board.getGameState()) {
+            if (square.isOccupied()) {
+                if (square.getOccupyingPiece().getColor() == color) {
+                    currentPlayersOccupiedSquares.add(square);
+                }
+            }
+        }
+    }
+
+    private void determineNonJumpMoves() {
+        for (Square startingSquare : this.currentPlayersOccupiedSquares) {
+            for (Square adjacentSquare : this.board.getAdjacentSquares(startingSquare)) {
+                Move normalMove = new Move(startingSquare.getPosition(),
+                        adjacentSquare.getPosition(), this.board);
+                if (MoveValidator.isValidMove(normalMove)) {
+                    this.possibleMoves.add(normalMove);
+                }
+            }
+        }
+    }
+
+    private void determineJumpMoves() {
+        for (Square startingSquare : this.currentPlayersOccupiedSquares) {
+            for (Square twoAway : this.board.getSquaresTwoMovesAway(startingSquare)) {
+                SingleJump jump = new SingleJump(startingSquare.getPosition(),
+                        twoAway.getPosition(), this.board);
+                if (MoveValidator.isValidMove(jump)) {
+                    ArrayList<SingleJump> jumps = new ArrayList<SingleJump>();
+                    jumps.add(jump);
+                    determineMultiJumpMoves(jumps);
+                }
+            }
+        }
+    }
+
+    private void determineMultiJumpMoves(ArrayList<SingleJump> jumps) {
```

Comment:
Great job on this. I knew this would be a tough method to write. Great use of recursion. :)